examples/gno.land/r/demo/users/users.gno

```diff
@@ -323,12 +325,26 @@ func Render(path string) string {
 	}
 }
 
-func renderHome() string {
+func renderHome(path string) string {
 	doc := ""
-	name2User.Iterate("", "", func(key string, value interface{}) bool {
-		user := value.(*users.User)
+
+	page := pager.NewPager(&name2User, 50).MustGetPageByPath(path)
+
+	for _, item := range page.Items {
+		user := item.Value.(*users.User)
 		doc += " * [" + user.Name + "](/r/demo/users:" + user.Name + ")\n"
-		return false
-	})
+	}
+
+	doc += page.Selector()
 	return doc
 }
```

Comment:
This doesn't work actually; if you try to make a pager with page size, say, 4, to check to pagination funcntionality, you get this;

![Screenshot 2024-10-08 at 15 04 23](https://github.com/user-attachments/assets/1b2dde7f-b8af-46d0-b8e0-4211d860c77a)

but then clicking on `2`, or `3` doesn't take you anywhere, just changes the link, as @jefft0 said. A click on 3 takes you here: `/r/demo/users?page=3`

Reply:
Yes, to have a working example, we depend on the other PR.

One option is to merge without using it everywhere, which means some contracts won't function fully for now. This is probably acceptable since we expect the other PR to be merged. We might consider disabling it on r/demo/users if people find the top-level listing too important on gnodev and portal loop (I don't share this opinion).

Alternatively, we need to ensure that someone works on #2876. This could be a core team member if the current contributor doesn't have time to finish.